t/geturl_connectivity.pl.in

```diff
@@ -55,35 +58,52 @@ sub run_httpd {
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '6', ssl => 0, curl => 0 },
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '6', ssl => 1, curl => 0 },
+    { ipv6_opt => 1, client_ipv => '4', server_ipv => '4', ssl => 0, curl => 0 },
+    { ipv6_opt => 1, client_ipv => '4', server_ipv => '4', ssl => 1, curl => 0 },
+    { ipv6_opt => 1, client_ipv => '6', server_ipv => '6', ssl => 0, curl => 0 },
+    { ipv6_opt => 1, client_ipv => '6', server_ipv => '6', ssl => 1, curl => 0 },
+    { ipv6_opt => 0, client_ipv => '',  server_ipv => '4', ssl => 0, curl => 1 },
+    { ipv6_opt => 0, client_ipv => '',  server_ipv => '4', ssl => 1, curl => 1 },
+    { ipv6_opt => 0, client_ipv => '4', server_ipv => '4', ssl => 0, curl => 1 },
+    { ipv6_opt => 0, client_ipv => '4', server_ipv => '4', ssl => 1, curl => 1 },
+    { ipv6_opt => 0, client_ipv => '6', server_ipv => '6', ssl => 0, curl => 1 },
+    { ipv6_opt => 0, client_ipv => '6', server_ipv => '6', ssl => 1, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '4', ssl => 0, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '4', ssl => 1, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '6', ssl => 0, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '',  server_ipv => '6', ssl => 1, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '4', server_ipv => '4', ssl => 0, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '4', server_ipv => '4', ssl => 1, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '6', server_ipv => '6', ssl => 0, curl => 1 },
+    { ipv6_opt => 1, client_ipv => '6', server_ipv => '6', ssl => 1, curl => 1 },
 );
 
 for my $tc (@test_cases) {
     SKIP: {
         skip("IO::Socket::INET6 not available", 1)
-            if ($tc->{ipv6_opt} || $tc->{client_ipv} eq '6') && !$has_io_socket_inet6;
+            if ($tc->{ipv6_opt} || $tc->{client_ipv} eq '6') && !$tc->{curl} && !$has_io_socket_inet6;
         skip("IPv6 not supported on this system", 1)
             if $tc->{server_ipv} eq '6' && !$ipv6_supported;
         skip("HTTP::Daemon too old for IPv6 support", 1)
             if $tc->{server_ipv} eq '6' && !$http_daemon_supports_ipv6;
         skip("HTTP::Daemon::SSL not available", 1) if $tc->{ssl} && !$has_http_daemon_ssl;
+        skip("Curl not available on this system", 1) if $tc->{curl} && !$has_curl;
         my $uri = $httpd{$tc->{server_ipv}}{$tc->{ssl} ? 'https' : 'http'}->endpoint();
-        my $name = sprintf("IPv%s client to %s (-%sipv6)",
-                           $tc->{client_ipv} || '*', $uri, $tc->{ipv6_opt} ? '' : 'no');
+        my $name = sprintf("IPv%s client to %s (-%sipv6)%s",
+                           $tc->{client_ipv} || '*', $uri, $tc->{ipv6_opt} ? '' : 'no',
+                           $tc->{curl} ? ' (curl)' : '');
         $ddclient::globals{'ipv6'} = $tc->{ipv6_opt};
+        $ddclient::globals{'curl'} = $tc->{curl};;
```

Comment:
If you default these to 0:

```suggestion
        $ddclient::globals{'ipv6'} = $tc->{ipv6_opt} // 0;
        $ddclient::globals{'curl'} = $tc->{curl} // 0;
```

then you can simplify the test cases:

```perl
    # Fetch via IO::Socket::INET or IO::Socket::SSL
    { ipv6_opt => 0, ... },
    ...
    # Fetch via IO::Socket::INET6 or IO::Socket::SSL
    { ipv6_opt => 1, ... },
    ...
    # Fetch via curl
    { curl => 1,     ... },
    ...
```

Reply:
Actually you would have to set the default earlier on so that the `skip` checks would work properly:

```perl
for my $tc (@test_cases) {
    $tc->{ipv6_opt} //= 0;
    $tc->{curl} //= 0;
    SKIP: {
        # ...
```

Reply:
Have set defaults before the skips.  Not sure about changing all the testcases to assume the defaults, but I did it anyway to see what you think.